src/internal/agent/hooks/flux-ocirepo.go

```diff
@@ -97,11 +103,34 @@ func mutateOCIRepo(ctx context.Context, r *v1.AdmissionRequest, cluster *cluster
 			patchedURL = fmt.Sprintf("%s:%s", patchedURL, src.Spec.Reference.Tag)
 		}
 
+		// Initially, we patch the src to include the crc32 hash
 		patchedSrc, err := transform.ImageTransformHost(registryAddress, patchedURL)
 		if err != nil {
 			return nil, fmt.Errorf("unable to transform the OCIRepo URL: %w", err)
 		}
 
+		timeoutCtx, cancel := context.WithTimeout(ctx, registryFetchTimeout)
+		defer cancel()
+
+		// Get the media type of the oci image
+		mediaType, err := getManifestConfigMediaType(timeoutCtx, zarfState, patchedSrc)
```

Comment:
@AustinAbro321 what are your thoughts on completeness of this solution and support of oci 1.1? 

I'd be willing to let this slide for now and writing an issue - but I believe the more complete solution would be primarily supporting `artifactType` and then falling back to `config.mediaType`. 

Thoughts?

Reply:
Could be as simple as:

```
mt := manifest.ArtifactType       // OCI 1.1+
if mt == "" {
    mt = manifest.Config.MediaType // pre-1.1 fallback
}
```

Reply:
This article does suggest that it's the way of the future https://oras.land/docs/concepts/artifact/, however Helm does not add an artifactType as of now. When they do add one it's fairly likely it won't be the same string as the media type so the check could start failing if they do add it. I think it's safer for now that we assume there's no artifact type then add it in the future